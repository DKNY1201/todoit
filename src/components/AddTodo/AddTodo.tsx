import * as React from 'react';

import {IconType, IconVersion} from '../../constants';
import Button from "../UI/Button/Button";
import Input from "../UI/Input/Input";
import './AddTodo.css';
import Icon from "../UI/Icon/Icon";
import {ITodo} from "../../model/todo";

interface IAddTodoState {
    isAdding: boolean;
    content: string;
    schedule?: string;
    labelId: number | undefined;
    priorityId?: number | undefined;
    parentTodoId?: number | undefined;
}

interface IAddTodoProps {
    labelId: number;
    addTodo: (todo: ITodo) => void;
}

const DEFAULT_PRIORITY = 4;

class AddTodo extends React.Component<IAddTodoProps, IAddTodoState> {
    state = {
        isAdding: false,
        content: '',
        schedule: '',
        labelId: -1,
        priorityId: DEFAULT_PRIORITY,
        parentTodoId: undefined,
    }

    public componentDidMount() {
        this.setState({
            labelId: this.props.labelId
        });
    }

    public render() {
        const {isAdding} = this.state;

        const addTodoContent = isAdding
            ? (
                <div className="add-todo-form">
                    <div className="add-todo-form-top">
                        <Input placeholder="Enter todo" changed={this.todoChangedHandler} />
                        <div className="add-todo-schedule" onClick={this.showScheduleModal}>schedule</div>
                    </div>
                    <div className="add-todo-form-bottom">
                        <div className="add-todo-form-bottom-left">
                            <Button label="Add Todo" style="highlight" clicked={this.addTodo} />
                            <Button label="Cancel" clicked={this.cancelAddTodo}/>
                        </div>
                        <div className="add-todo-form-bottom-right">
                            <Icon type={IconType.Action} version={IconVersion.Fas} clicked={this.showLabelModal} />
                            <Icon type={IconType.Action} version={IconVersion.Fas} clicked={this.showPriorityModal} />
                        </div>
                    </div>
                </div>
            )
            : (
                <div className="add-todo-text" onClick={this.triggerAdding}>
                    <span>Add Todo</span>
                </div>
            );

        return (
            <div className="add-todo-container">
                {addTodoContent}
            </div>
        );
    }

    private todoChangedHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({content: e.target.value});
    }

    private triggerAdding = () => {
        this.setState({
            isAdding: true
        });
    }

    private showScheduleModal = () => {
        console.log('showScheduleModal');
    }

    private showLabelModal = () => {
        console.log('showLabelModal');
    }

    private showPriorityModal = () => {
        console.log('showPriorityModal');
    }

    private addTodo = () => {
        const {content, schedule, labelId, priorityId, parentTodoId} = this.state;
        const newTodo: ITodo = {
            content,
            schedule,
            labelId,
            priorityId,
            parentTodoId
        };
        this.props.addTodo(newTodo);
    }

    private cancelAddTodo = () => {
        console.log('cancelAddTodo');
    }
}

export default AddTodo;