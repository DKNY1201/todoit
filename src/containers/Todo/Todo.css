.todo-container {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 10px;
}

.todo-container .todo-central {
    display: flex;
    justify-content: center;
    align-items: center;
}