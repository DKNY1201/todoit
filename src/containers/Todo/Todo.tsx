import * as React from 'react';

import Checkbox from '../../components/UI/Checkbox/Checkbox';
import Icon from '../../components/UI/Icon/Icon';
import './Todo.css';

export interface ITodoProp {
	completeTodo: () => void;
	content: string;
	schedule?: string;
}

interface ITodoState {
	isHovering: boolean;
}

class Todo extends React.Component<ITodoProp, ITodoState> {
	state = {
		isHovering: false
	}


	public render(): React.ReactNode {
		const {isHovering} = this.state;
		const {completeTodo, content, schedule} = this.props;

		return (
			<div className="todo-container" onMouseEnter={this.toggleIsHover} onMouseLeave={this.toggleIsHover}>
				{ isHovering && <Icon type="drag-drop" /> }
				<div className="todo-central">
					<Checkbox clicked={completeTodo} />
					<span className="todo-content">{content}</span>
					{ isHovering && <Icon type="comment" clicked={this.showCommentModal} /> }
					{
						!(!schedule && !isHovering) && <div className="todo-schedule">
							{ schedule ? schedule : isHovering ? <Icon type="schedule" clicked={this.showScheduleModal} /> : ''}
						</div>
					}


				</div>
				{ isHovering && <Icon type="action" clicked={this.showActionModal} /> }
			</div>
		);
	}

	private showCommentModal = () => {
		console.log('showCommentModal');
	}

	private showScheduleModal = () => {
		console.log('showScheduleModal');
	}

	private showActionModal = () => {
		console.log('showActionModal');
	}

	private toggleIsHover = () => {
		this.setState((prevState) => {
			return {
				isHovering:!prevState.isHovering
			}
		})
	}
}

// Todo.propTypes = {
// 	completeTodo: PropTypes.func.isRequired,
// 	content: PropTypes.string.isRequired,
// 	schedule: PropTypes.string.isRequired,
// };

export default Todo;