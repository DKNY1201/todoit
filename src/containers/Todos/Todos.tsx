import * as React from 'react';
import {connect} from 'react-redux';

import AddTodo from '../../components/AddTodo/AddTodo';
import Todo from '../../components/Todo/Todo';
import {IState, ITodoState} from "../../types/index";
import './Todos.css';

export interface ITodosState {
    isAddingTodo: boolean;
}

export class Todos extends React.Component<ITodoState, ITodosState> {
    state = {
        isAddingTodo: false
    };


    render() {
        const {todos, labels} = this.props;
        const defaultLabel = {
            id: 1,
            color: '#990000',
            title: 'Personal',
            parentLabelId: undefined,
            isFavorite: false,
            isSelected: true
        }
        const selectedLabel = labels.find(label => label.isSelected) || defaultLabel;
        const displayName = selectedLabel ? selectedLabel.title : 'N/A';

        const todoItems = todos
            .filter(todo => !todo.isCompleted)
            .filter(todo => todo.labelId === selectedLabel.id)
            .map(todo => <Todo key={todo.id} />);

        return (
            <div className="todoContainer">
                <h1 className="displayName">{displayName}</h1>
                {todoItems}
                <AddTodo />
            </div>
        )
    }
}


const mapStateToProps = ({todo}: IState) => {
    const {labels, priorities, todos} = todo;
    return {
        labels,
        priorities,
        todos,
    }
}

// const mapDispatchToProps = dispatch => {
// 	return {
// 		onIngredientAdded: (igName) => dispatch(actions.addIngredient(igName)),
//
// 	}
// }
export default connect(mapStateToProps)(Todos);