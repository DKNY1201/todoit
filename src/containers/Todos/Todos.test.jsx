import * as React from 'react';
import * as enzyme from 'enzyme';

import Todos from './Todos';
import Todo from '../../components/Todo/Todo';
import AddTodo from '../../components/AddTodo/AddTodo';

describe('<Todos />', () => {
	let wrapper, wrapperInstance;

	// beforeEach(() => {
	// 	const labels = [
	// 		{
	// 			id: 1,
	// 			color: '#990000',
	// 			title: 'Personal',
	// 			parentLabelId: null,
	// 			isFavorite: false,
	// 			isSelected: false
	// 		},
	// 		{
	// 			id: 2,
	// 			color: '#000000',
	// 			title: 'Work',
	// 			parentLabelId: null,
	// 			isFavorite: false,
	// 			isSelected: true
	// 		}
	// 	];
	//
	// 	const priorities = [
	// 		{
	// 			id: 1,
	// 			title: 'Priority 1',
	// 			icon: 'flag-red',
	// 			isSelected: false
	// 		},
	// 		{
	// 			id: 2,
	// 			title: 'Priority 2',
	// 			icon: 'flag-orange',
	// 			isSelected: false
	// 		}
	// 	];
	//
	// 	const todos = [
	// 		{
	// 			id: 1,
	// 			title: 'Buy Skateboard',
	// 			description: 'Target, Home Depot or Walmart',
	// 			time: '03/20/2019',
	// 			labelId: 1,
	// 			priorityId: 2,
	// 			userId: 1,
	// 			parentTodoId: null,
	// 			isCompleted: false
	// 		},
	// 		{
	// 			id: 2,
	// 			title: 'Learn to Skateboard',
	// 			description: 'Watch Youtube, find a community',
	// 			time: '05/16/2019',
	// 			labelId: 2,
	// 			priorityId: 2,
	// 			userId: 1,
	// 			parentTodoId: null,
	// 			isCompleted: false
	// 		},
	// 		{
	// 			id: 3,
	// 			title: 'Learn GraphQL',
	// 			description: 'Follow the course on jscomplete.com',
	// 			time: '07/15/2019',
	// 			labelId: 2,
	// 			priorityId: 3,
	// 			userId: 1,
	// 			parentTodoId: null,
	// 			isCompleted: false
	// 		},
	// 		{
	// 			id: 4,
	// 			title: 'Learn GraphQL 1',
	// 			description: 'Follow the course on jscomplete.com 1',
	// 			time: '07/15/2019',
	// 			labelId: 2,
	// 			priorityId: 3,
	// 			userId: 1,
	// 			parentTodoId: null,
	// 			isCompleted: true
	// 		}
	// 	];
	//
	// 	// wrapper.setProps({labels, priorities, todos});
	// 	wrapper = enzyme.shallow(<Todos labels={labels} priorities={priorities} todos={todos} />);
	// 	wrapperInstance = wrapper.instance();
	// });

	it('should always render display name', () => {
		const labels = [
			{
				id: 1,
				color: '#990000',
				title: 'Personal',
				parentLabelId: null,
				isFavorite: false,
				isSelected: false
			},
			{
				id: 2,
				color: '#000000',
				title: 'Work',
				parentLabelId: null,
				isFavorite: false,
				isSelected: true
			}
		];

		const priorities = [
			{
				id: 1,
				title: 'Priority 1',
				icon: 'flag-red',
				isSelected: false
			},
			{
				id: 2,
				title: 'Priority 2',
				icon: 'flag-orange',
				isSelected: false
			}
		];

		const todos = [
			{
				id: 1,
				title: 'Buy Skateboard',
				description: 'Target, Home Depot or Walmart',
				time: '03/20/2019',
				labelId: 1,
				priorityId: 2,
				userId: 1,
				parentTodoId: null,
				isCompleted: false
			},
			{
				id: 2,
				title: 'Learn to Skateboard',
				description: 'Watch Youtube, find a community',
				time: '05/16/2019',
				labelId: 2,
				priorityId: 2,
				userId: 1,
				parentTodoId: null,
				isCompleted: false
			},
			{
				id: 3,
				title: 'Learn GraphQL',
				description: 'Follow the course on jscomplete.com',
				time: '07/15/2019',
				labelId: 2,
				priorityId: 3,
				userId: 1,
				parentTodoId: null,
				isCompleted: false
			},
			{
				id: 4,
				title: 'Learn GraphQL 1',
				description: 'Follow the course on jscomplete.com 1',
				time: '07/15/2019',
				labelId: 2,
				priorityId: 3,
				userId: 1,
				parentTodoId: null,
				isCompleted: true
			}
		];
		const wrapper = enzyme.shallow(<Todos labels={labels} priorities={priorities} todos={todos} />);
		// expect(wrapper.find('h1').length).toEqual(1);
		expect(wrapper.find('h1').first().text()).not.toEqual('');
	});

	it('should render selected label name', () => {
		expect(wrapper.find('h1.displayName').text()).toEqual('Work');
	});

	it('should render selected priority name', () => {
		const priorities = [
			{
				id: 1,
				title: 'Priority 1',
				icon: 'flag-red',
				isSelected: false
			},
			{
				id: 2,
				title: 'Priority 2',
				icon: 'flag-orange',
				isSelected: true
			}
		];

		wrapper.setProps({priorities});
		expect(wrapper.find('h1.displayName').text()).toEqual('Priority 2');
	});

	it('should render list of Todo component that are not completed and under selected label', () => {
		expect(wrapper.find(Todo).length).toEqual(2);
	});

	it('should render a AddTodo component', () => {
		expect(wrapper.find(AddTodo).length).toEqual(1);
	});
});

describe('mounted <Todos />', () => {

});