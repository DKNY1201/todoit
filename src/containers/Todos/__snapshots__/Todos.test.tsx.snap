// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todos /> rendering should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todos
    labels={
        Array [
            Object {
              "color": "#990000",
              "id": 1,
              "isFavorite": false,
              "isSelected": false,
              "parentLabelId": null,
              "title": "Personal",
            },
            Object {
              "color": "#000000",
              "id": 2,
              "isFavorite": false,
              "isSelected": true,
              "parentLabelId": null,
              "title": "Work",
            },
          ]
    }
    priorities={
        Array [
            Object {
              "icon": "flag-red",
              "id": 1,
              "isSelected": false,
              "title": "Priority 1",
            },
            Object {
              "icon": "flag-orange",
              "id": 2,
              "isSelected": false,
              "title": "Priority 2",
            },
          ]
    }
    todos={
        Array [
            Object {
              "description": "Target, Home Depot or Walmart",
              "id": 1,
              "isCompleted": false,
              "labelId": 1,
              "parentTodoId": null,
              "priorityId": 2,
              "time": "03/20/2019",
              "title": "Buy Skateboard",
              "userId": 1,
            },
            Object {
              "description": "Watch Youtube, find a community",
              "id": 2,
              "isCompleted": false,
              "labelId": 2,
              "parentTodoId": null,
              "priorityId": 2,
              "time": "05/16/2019",
              "title": "Learn to Skateboard",
              "userId": 1,
            },
            Object {
              "description": "Follow the course on jscomplete.com",
              "id": 3,
              "isCompleted": false,
              "labelId": 2,
              "parentTodoId": null,
              "priorityId": 3,
              "time": "07/15/2019",
              "title": "Learn GraphQL",
              "userId": 1,
            },
            Object {
              "description": "Follow the course on jscomplete.com 1",
              "id": 4,
              "isCompleted": true,
              "labelId": 2,
              "parentTodoId": null,
              "priorityId": 3,
              "time": "07/15/2019",
              "title": "Learn GraphQL 1",
              "userId": 1,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="label-title"
>
          Work
</h1>,
        Array [
          <Todo
            completeTodo={[Function]}
            content={undefined}
            schedule={undefined}
/>,
          <Todo
            completeTodo={[Function]}
            content={undefined}
            schedule={undefined}
/>,
        ],
        <AddTodo
          addTodo={[Function]}
          labelId={2}
/>,
      ],
      "className": "todos-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Work",
          "className": "label-title",
        },
        "ref": null,
        "rendered": "Work",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "completeTodo": [Function],
          "content": undefined,
          "schedule": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "completeTodo": [Function],
          "content": undefined,
          "schedule": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addTodo": [Function],
          "labelId": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="label-title"
>
            Work
</h1>,
          Array [
            <Todo
              completeTodo={[Function]}
              content={undefined}
              schedule={undefined}
/>,
            <Todo
              completeTodo={[Function]}
              content={undefined}
              schedule={undefined}
/>,
          ],
          <AddTodo
            addTodo={[Function]}
            labelId={2}
/>,
        ],
        "className": "todos-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Work",
            "className": "label-title",
          },
          "ref": null,
          "rendered": "Work",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "completeTodo": [Function],
            "content": undefined,
            "schedule": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "completeTodo": [Function],
            "content": undefined,
            "schedule": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addTodo": [Function],
            "labelId": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
